selfdrive/ui/qt/maps/map.cc

```diff
@@ -375,16 +379,17 @@ void MapWindow::offroadTransition(bool offroad) {
   last_bearing = {};
 }
 
-void MapWindow::updateDestinationMarker(std::optional<QMapbox::Coordinate> nav_dest) {
-  if (nav_dest) {
+void MapWindow::updateDestinationMarker() {
+  m_map->setPaintProperty("pinLayer", "visibility", "none");
+
+  auto nav_dest = coordinate_from_param("NavDestination");
```

Comment:
should we use last_valid_nav_dest here, instead of read it  from param?

Reply:
if you get a navRoute with no coords, the value of last_valid_nav_dest will be different than reading from the param